plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/transport/http/OpenLineageHttpTransport.java

```diff
@@ -77,4 +61,11 @@ public HttpTransport buildTransport()
                         this.headers,
                         null));
     }
+
+    private static ApiKeyTokenProvider createToken(String token)
+    {
+        ApiKeyTokenProvider apiKeyTokenProvider = new ApiKeyTokenProvider();
+        apiKeyTokenProvider.setApiKey(token);
+        return apiKeyTokenProvider;
```

Comment:
`ApiKeyTokenProvider` is mutable. We should avoid putting such object in fields. 

I think it would be better to a) changing `ApiKeyTokenProvider apiKey` to `Optional<String>` and build the token provider in `buildTransport` or b) just staying with the current implementation. 

If we choose b), I prefer the previous private static class to record so that we can avoid having both `token()` and `getToken()` method. The record isn't good from method naming consistency in this case in my opinion.

Reply:
`ApiKeyTokenProvider` is a class provided by OpenLineage itself.

Reply:
I think we could lazily create this `ApiKeyTokenProvider` when we are building the `HttpConfig` - since it is mutable in the runtime we are not sure if some of the internal class could invoke the `setApiKey` in the future. I'm inclined toward option -b with `static class` instead of `record`. WDYT ?

Reply:
I have a third option. PTAL.

Reply:
No class, no methods, immutable.

Reply:
Sounds great !!